### PR TITLE
Add ADR-0020: Minimum supported Python version

### PR DESCRIPTION
**Pull request by frenck**
This adds ADR-0020: Minimum supported Python version

It replaces ADR-0002.

Discussion: https://github.com/home-assistant/architecture/discussions/903

### CODE REVIEW

**Review by bdraco — APPROVED** [review]

**Review by balloob — APPROVED** [review]

File: adr/0020-minimum-supported-python-version copy.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# 0020. Minimum supported Python version
```
Comment (ludeeus) [inline adr/0020-minimum-supported-python-version copy.md:1]: ` copy` seems like it should not be in the filename.
Needs adjustment in 0002 if changed

Comment (frenck) [reply adr/0020-minimum-supported-python-version copy.md:1]: We've discussed this, as this is a pretty different approach; this is no longer amending but fundamentally changing. Hence, a new ADR is created, and the old one is updated and marked as superseded, while a new one is introduced.

Comment (frenck) [reply adr/0020-minimum-supported-python-version copy.md]: Pushed a fix for the filename 👍 Thanks for spotting 👀

Comment (ludeeus) [reply adr/0020-minimum-supported-python-version copy.md]: It was the `%20copy` part of the link in 0002 I was referring to

Comment (frenck) [reply adr/0020-minimum-supported-python-version copy.md]: Fixed that one as well 👍

**Review by ludeeus — APPROVED** [review]

**Review by Jc2k — APPROVED** [review]

File: adr/0020-minimum-supported-python-version.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+## Decision
+
+Home Assistant supports a single minor version of Python at a time, with the goal of supporting the latest minor upstream Python version.
+
+When a new minor Python version is released, the Home Assistant project will provide compatibility and support for that new version as soon as possible.
+
+Once the new version is supported by the Home Assistant project and rolled out in our container distributions, the previous Python version will be deprecated for a period of two release cycles. Home Assistant will print a warning message during the deprecation period to inform the user.
+
+Development during the deprecation period will take place on the newer Python version. We rely on automated tests and linting to catch backward incompatible changes. The deprecated version will stay in our CI for an additional release cycle (that is a total of three),
+to accommodate the patch releases that belong last and previous release of the deprecation period.
```
[the remark this answers is not in the document]
Comment (frenck) [reply adr/0020-minimum-supported-python-version.md:26]: Thanks! #914

File: adr/0020-minimum-supported-python-version.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+The minimal supported micro/patch Python version is decided on a minor version basis and codified in the version check code in Home Assistant proper.
+
+## Consequences
+
+We have three installation methods that are based on Docker containers (Home Assistant OS, Container, and Supervised), which are maintained and kept up to date by the Home Assistant project. A significant majority of our user base uses these installation methods. For users using the Core installation method (running their own a Python environment), they need to upgrade their Python version at least once a year.
```
[the remark this answers is not in the document]
Comment (frenck) [reply adr/0020-minimum-supported-python-version.md:32]: Thanks! #914

**Pull request merged**